### PR TITLE
infra: add CI job for running tutorial notebooks

### PR DESCRIPTION
**Pull request by ccurme**

### CODE REVIEW

File: .github/workflows/run_notebooks.yml
Comment (efriis) [inline .github/workflows/run_notebooks.yml:9]: might be worth parameterizing this to make sure they run in 3.9 + 3.12 mostly

File: .github/workflows/run_notebooks.yml
Comment (efriis) [inline .github/workflows/run_notebooks.yml:58]: if we know they won't be needed, could we skip populating the real secrets and instead just run `export ANTHROPIC_API_KEY=fake-api-key` commands?

File: docs/scripts/prepare_notebooks_for_ci.py
Comment (efriis) [inline docs/scripts/prepare_notebooks_for_ci.py:22]: ooc why can't this be run off a cassette?

File: docs/scripts/execute_notebooks.sh
Comment (efriis) [inline docs/scripts/execute_notebooks.sh:11]: might be worth switching to `jupyter nbconvert --to notebook --execute $file` for later customizability if we ever want to process the outputs / look at the file

doesn't need to block this PR - this looks like it only gets errors from the jupyter runtime and might be tricky to get warnings/incorrect outputs from cells?

File: docs/scripts/execute_notebooks.sh
Comment (efriis) [inline docs/scripts/execute_notebooks.sh:26]: if there's a way to parameterize this, might make this CI step more customizable in future if we want to use it to run e.g. individual notebooks

**Review by efriis — APPROVED** [review]

File: .github/workflows/run_notebooks.yml
Comment (ccurme) [reply .github/workflows/run_notebooks.yml:9]: Will do separately as it's easier to debug workflow dispatch after merging the new workflow to master.

File: .github/workflows/run_notebooks.yml
Comment (ccurme) [reply .github/workflows/run_notebooks.yml]: Deleted this comment as these actually do get used for notebooks where we skip cassettes.

File: docs/scripts/prepare_notebooks_for_ci.py
Comment (ccurme) [reply docs/scripts/prepare_notebooks_for_ci.py:22]: Local LLM notebook I think may not make http calls, there's just significant overhead running local models (e.g., ollama). LMK if I've misunderstood.

File: docs/scripts/execute_notebooks.sh
Comment (ccurme) [reply docs/scripts/execute_notebooks.sh:26]: Will look into this after merging so we can debug workflow dispatch.

File: .github/workflows/run_notebooks.yml
Comment (ccurme) [reply .github/workflows/run_notebooks.yml:9]: Added python version as a parameter in https://github.com/langchain-ai/langchain/pull/26956

File: docs/scripts/execute_notebooks.sh
Comment (ccurme) [reply docs/scripts/execute_notebooks.sh:26]: Added in https://github.com/langchain-ai/langchain/pull/26956

**Pull request merged**
